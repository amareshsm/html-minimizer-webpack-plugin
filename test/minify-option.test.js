import HtmlMinimizerPlugin from '../src/index';

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} from './helpers';

describe('"minify" option', () => {
  it('should work minify function', async () => {
    const testHtmlId = './simple.html';
    const compiler = getCompiler(testHtmlId);

    new HtmlMinimizerPlugin({
      minimizerOptions: {
        collapseWhitespace: true,
      },
      minify: (data, minimizerOptions) => {
        // eslint-disable-next-line global-require
        const htmlMinifier = require('html-minifier-terser');
        const [[, input]] = Object.entries(data);

        return htmlMinifier.minify(input, minimizerOptions);
      },
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.html$/i)).toMatchSnapshot('assets');
    expect(getErrors(stats)).toMatchSnapshot('errors');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
  });
});
