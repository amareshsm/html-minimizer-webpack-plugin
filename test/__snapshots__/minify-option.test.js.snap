// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"minify" option should work if minify is array && minimizerOptions is array: assets 1`] = `
Object {
  "simple.html": "<!doctype html> <html lang=\\"en\\"> <head> <meta charset=\\"UTF-8\\"> <meta name=\\"viewport\\" content=\\"width=device-width,user-scalable=no,initial-scale=1,maximum-scale=1,minimum-scale=1\\"> <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> <ul> <li>Coffee</li> <li>Tea</li> <li>Milk</li> </ul> <h2>An Ordered HTML List</h2> <ol> <li>Coffee</li> <li>Tea</li> <li>Milk</li> </ol> </body> </html> <div>Second function: text from option</div>",
}
`;

exports[`"minify" option should work if minify is array && minimizerOptions is array: errors 1`] = `Array []`;

exports[`"minify" option should work if minify is array && minimizerOptions is array: warnings 1`] = `Array []`;

exports[`"minify" option should work if minify is array && minimizerOptions is object: assets 1`] = `
Object {
  "simple.html": "<!doctype html> <html lang=\\"en\\"> <head> <meta charset=\\"UTF-8\\"> <meta name=\\"viewport\\" content=\\"width=device-width,user-scalable=no,initial-scale=1,maximum-scale=1,minimum-scale=1\\"> <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> <ul> <li>Coffee</li> <li>Tea</li> <li>Milk</li> </ul> <h2>An Ordered HTML List</h2> <ol> <li>Coffee</li> <li>Tea</li> <li>Milk</li> </ol> </body> </html> <div>Second function: true</div><div>Third function: true</div>",
}
`;

exports[`"minify" option should work if minify is array && minimizerOptions is object: errors 1`] = `Array []`;

exports[`"minify" option should work if minify is array && minimizerOptions is object: warnings 1`] = `Array []`;

exports[`"minify" option should work minify function: assets 1`] = `
Object {
  "simple.html": "<!doctype html><html lang=\\"en\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,user-scalable=no,initial-scale=1,maximum-scale=1,minimum-scale=1\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><title>Document</title></head><body><h1>My First Heading</h1><p>My first paragraph.</p><h2>An Unordered HTML List</h2><ul><li>Coffee</li><li>Tea</li><li>Milk</li></ul><h2>An Ordered HTML List</h2><ol><li>Coffee</li><li>Tea</li><li>Milk</li></ol></body></html>",
}
`;

exports[`"minify" option should work minify function: errors 1`] = `Array []`;

exports[`"minify" option should work minify function: warnings 1`] = `Array []`;
